[pt1][quant] Unit Test for the Dynamic Quantized Linear operator

jianyuh opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#23139 [pt1][quant] Unit Test for the Dynamic Quantized Linear operator**

Add a unit test for the Dynamic Quantized Linear operator (```torch.fbgemm_linear_quantize_weight```, ```torch.fbgemm_pack_quantized_matrix```, and ```torch.fbgemm_linear_int8_weight```) in ```test_quantized.py```.

Differential Revision: [D16404027](https://our.internmc.facebook.com/intern/diff/D16404027/)

jianyuh committed:
[WIP][pt1][quant] Unit Test for the Dynamic Quantized Linear operator (torch.fbgemm_linear_quantize_weight, torch.fbgemm_pack_quantized_matrix, and torch.fbgemm_linear_int8_weight) in test_quantized.py

Add a unit test for the Dynamic Quantized Linear operator (```torch.fbgemm_linear_quantize_weight```, ```torch.fbgemm_pack_quantized_matrix```, and ```torch.fbgemm_linear_int8_weight```) in ```test_quantized.py```.

Differential Revision: [D16404027](https://our.internmc.facebook.com/intern/diff/D16404027/)

jianyuh committed:
Update on "[WIP][pt1][quant] Unit Test for the Dynamic Quantized Linear operator"


Add a unit test for the Dynamic Quantized Linear operator (```torch.fbgemm_linear_quantize_weight```, ```torch.fbgemm_pack_quantized_matrix```, and ```torch.fbgemm_linear_int8_weight```) in ```test_quantized.py```.

Differential Revision: [D16404027](https://our.internmc.facebook.com/intern/diff/D16404027/)

zafartahirov commented on test/test_quantized.py:
```diff
@@ -315,6 +315,27 @@ def test_qlinear(self, use_bias, use_relu):
 
         torch.testing.assert_allclose(Y_fp32, Y_fp32_ref, rtol=0.0001, atol=1e-3)
 
+    def test_dynamic_qlinear(self):
```
Hypothesis instead of randn?

jianyuh replied:
Fixed.

jianyuh committed:
Update on "[WIP][pt1][quant] Unit Test for the Dynamic Quantized Linear operator"


Add a unit test for the Dynamic Quantized Linear operator (```torch.fbgemm_linear_quantize_weight```, ```torch.fbgemm_pack_quantized_matrix```, and ```torch.fbgemm_linear_int8_weight```) in ```test_quantized.py```.

Differential Revision: [D16404027](https://our.internmc.facebook.com/intern/diff/D16404027/)

raghuramank100 commented on test/test_quantized.py:
```diff
@@ -417,6 +417,93 @@ def test_qlinear(self, batch_size, input_channels, output_channels, use_bias, us
+        input_channels=st.integers(16, 32),
+        output_channels=st.integers(4, 8),
+    )
+    def test_qlinear_legacy(self, batch_size, input_channels, output_channels):
```
Is this a dup? I see the same code under test_qlinear in the same file in the other diff, please update

jianyuh replied:
@raghuramank100 : This is for the unit test of the original `torch.fbgemm_linear_quantize_weight`, `torch.fbgemm_pack_quantized_matrix`, and `torch.fbgemm_linear_int8_weight` APIs for `torch.jit.quantized` workflow.

jianyuh committed:
Update on "[pt1][quant] Unit Test for the Dynamic Quantized Linear operator"


Add a unit test for the Dynamic Quantized Linear operator (```torch.fbgemm_linear_quantize_weight```, ```torch.fbgemm_pack_quantized_matrix```, and ```torch.fbgemm_linear_int8_weight```) in ```test_quantized.py```.

Differential Revision: [D16404027](https://our.internmc.facebook.com/intern/diff/D16404027/)

jianyuh committed:
Update on "[pt1][quant] Unit Test for the Dynamic Quantized Linear operator"


Add a unit test for the Dynamic Quantized Linear operator (```torch.fbgemm_linear_quantize_weight```, ```torch.fbgemm_pack_quantized_matrix```, and ```torch.fbgemm_linear_int8_weight```) in ```test_quantized.py```.

Differential Revision: [D16404027](https://our.internmc.facebook.com/intern/diff/D16404027/)

jianyuh committed:
Update on "[pt1][quant] Unit Test for the Dynamic Quantized Linear operator"


Add a unit test for the Dynamic Quantized Linear operator (```torch.fbgemm_linear_quantize_weight```, ```torch.fbgemm_pack_quantized_matrix```, and ```torch.fbgemm_linear_int8_weight```) in ```test_quantized.py```.

Differential Revision: [D16404027](https://our.internmc.facebook.com/intern/diff/D16404027/)

jianyuh committed:
Update on "[pt1][quant] Unit Test for the Dynamic Quantized Linear operator"


Add a unit test for the Dynamic Quantized Linear operator (```torch.fbgemm_linear_quantize_weight```, ```torch.fbgemm_pack_quantized_matrix```, and ```torch.fbgemm_linear_int8_weight```) in ```test_quantized.py```.

Differential Revision: [D16404027](https://our.internmc.facebook.com/intern/diff/D16404027/)

jamesr66a commented:
@jianyuh do you want to land this?

jianyuh commented:
@jamesr66a Either works. If we still want to keep track of the legacy QuantizedLinear ops (and if you approve this PR), we can land this and make sure the unit tests can cover more cases. Otherwise, we can just close this PR.

jamesr66a reviewed (approved):
Seems fine, more test coverage probably can't hurt

jianyuh committed:
Update on "[pt1][quant] Unit Test for the Dynamic Quantized Linear operator"


Add a unit test for the Dynamic Quantized Linear operator (```torch.fbgemm_linear_quantize_weight```, ```torch.fbgemm_pack_quantized_matrix```, and ```torch.fbgemm_linear_int8_weight```) in ```test_quantized.py```.

Differential Revision: [D16404027](https://our.internmc.facebook.com/intern/diff/D16404027/)

[ghstack-poisoned]

jianyuh committed:
Update on "[pt1][quant] Unit Test for the Dynamic Quantized Linear operator"


Add a unit test for the Dynamic Quantized Linear operator (```torch.fbgemm_linear_quantize_weight```, ```torch.fbgemm_pack_quantized_matrix```, and ```torch.fbgemm_linear_int8_weight```) in ```test_quantized.py```.

Differential Revision: [D16404027](https://our.internmc.facebook.com/intern/diff/D16404027/)

[ghstack-poisoned]

jianyuh committed:
Update on "[pt1][quant] Unit Test for the Dynamic Quantized Linear operator"


Add a unit test for the Dynamic Quantized Linear operator (```torch.fbgemm_linear_quantize_weight```, ```torch.fbgemm_pack_quantized_matrix```, and ```torch.fbgemm_linear_int8_weight```) in ```test_quantized.py```.

Differential Revision: [D16404027](https://our.internmc.facebook.com/intern/diff/D16404027/)

[ghstack-poisoned]
